pinot-common/src/main/java/org/apache/pinot/common/protocols/SegmentCompletionProtocol.java

```diff
@@ -200,6 +203,7 @@ public String getUrl(String hostPort, String protocol) {
       private String _segmentLocation;
       private long _memoryUsedBytes;
       private long _segmentSizeBytes;
+      private SegmentMetadata _segmentMetadata;
```

Comment:
You should not be sending segment metadata as a query parameter.

Reply:
Done.